config/src/main/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LoginConfigurer.java

```diff
@@ -103,6 +103,20 @@
 
 	private RelyingPartyRegistrationRepository relyingPartyRegistrationRepository;
 
+	private AuthenticationProvider authenticationProvider;
+
+	/**
+	 * Allows a configuration of a {@link AuthenticationProvider} to be used during SAML 2 authentication.
+	 * If none is specified, the system will create one and insert it into the ProviderManager list
+	 * @param authenticationProvider the authentication provider to be used
+	 * @return the {@link Saml2LoginConfigurer} for further configuration
+	 * @since 5.2.2
+	 */
+	public Saml2LoginConfigurer authenticationProvider(AuthenticationProvider authenticationProvider) {
```

Comment:
Instead of exposing an `authenticationProvider(..)` configuration option, we may want to consider providing `authoritiesExtractor(...)` and `authoritiesMapper(...)` under `saml2Login()` OR under a _nested config class_. This would be consistent with `oauth2Login()`, `oauth2Client()` and `oauth2ResourceServer()` as neither expose `authenticationProvider()` and instead handle the configuration of the provider in `configure(http)`.

Reply:
Thanks @jgrandja . This was my first consideration. That leaves the implementation with an inability to affect or easily swap out SAML response parsing and validation, something that can definitely change between providers. The idea here was to follow Spring logic that one should be able to swap out each component, not just the values of a component.

Reply:
I've also added in a _spec_ for configuring the defaults settings.

Reply:
This was removed. A post processor approach has been documented in the SAML documentation